interception.go

```diff
@@ -64,21 +77,37 @@ func newInterceptionProcessor(p Provider, logger slog.Logger, recorder Recorder,
 			return
 		}
 
+		route := strings.TrimPrefix(r.URL.Path, fmt.Sprintf("/%s", p.Name()))
 		log := logger.With(
-			slog.F("route", r.URL.Path),
+			slog.F("route", route),
 			slog.F("provider", p.Name()),
 			slog.F("interception_id", interceptor.ID()),
 			slog.F("user_agent", r.UserAgent()),
+			slog.F("streaming", interceptor.Streaming()),
 		)
 
 		log.Debug(r.Context(), "interception started")
+		if metrics != nil {
```

Comment:
ok, it is confusing to see `if metrics != nil` multiple times, should we do something about it?

Reply:
Metrics are optional and has to be a pointer type. Could make this an interface type but it seems unnecessary for now. I agree it ain't pretty.